cmd/kubectl_token.go

```diff
@@ -671,30 +662,19 @@ func getTLSConfig(skipVerify bool, caCerts string) (*tls.Config, error) {
 	return config, nil
 }
 
-func request(method, url string, body io.Reader) ([]byte, error) {
-	var response []byte
-
-	req, err := http.NewRequest(method, url, body)
+func doRequest(client *http.Client, req *http.Request) (*http.Response, []byte, error) {
```

Comment:
Return the `http.Response` as the consuming code should be checking the returned status code in most/all cases.